### PR TITLE
blocklist dual sign off for multi-guid submission

### LINKED ISSUE #13201: implement dual sign-off for multi block add/edit
part of #13089 

The current read-only view for multiblocks needs updating to allow blocks to be confirmed.  Changes should not now be immediately applied.  In general anyway because 
> there could be a threshold for requiring dual sign-off. If all add-ons involved in a block have user counts under that threshold, the block change would be allowed without sign-off.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #13201 which is part of #13089 

This patch transforms the `MultiBlockSubmit` class into `BlockSubmission`, so it can handle a sign-off of the guids being submitted before the cron job handles the Blocks being created.

Because this is already quite a large series of commits I've deferred the following requested functionality to future prs/issues:
- log who signed off in Block history when the Block is created
- show/block further edits in existing Block page when a `BlockSubmission` is outstanding
- separate permissions for signoff & submission
- replace the dual sign-off waffle with a condition around addon ADU

Additionally, there was some other functionality I'd like to add that wasn't explicitly requested:
- show in GUID input too if a matching GUID is in an outstanding `BlockSubmission`
- allow rejected `BlockSubmission`'s to be edited or re-used/resubmitted easily
- rewrite `BlockSubmission.add_view` to use more vanilla `ModelAdmin` functionality like `change_view` does; use same templates and style sheets, etc.

### CODE REVIEW

File: src/olympia/blocklist/admin.py [hunk 1]
```diff
@@ -141,14 +129,75 @@ class MultiBlockSubmitAdmin(admin.ModelAdmin):
...
+    change_form_template = 'blocklist/block_submission_change_form.html'
+
+    class Media:
+        css = {
+            'all': ('css/admin/blocklist_blocksubmission.css',)
+        }
 
     def has_delete_permission(self, request, obj=None):
+        # For now, keep all BlockSubmission records.
+        # TODO: define under what cirumstances records can be safely deleted.
```
Comment (willdurand) [inline src/olympia/blocklist/admin.py]: Is there an issue filed for this `TODO`? If not, I suggest creating one.

File: src/olympia/blocklist/admin.py [hunk 2]
```diff
@@ -141,14 +129,75 @@ class MultiBlockSubmitAdmin(admin.ModelAdmin):
...
     def has_change_permission(self, request, obj=None):
-        return False
+        has = super().has_change_permission(request, obj=obj)
+        pending = obj and obj.signoff_state == BlockSubmission.SIGNOFF_PENDING
+        return has and (not obj or pending)
+
+    def has_signoff_permission(self, request, obj=None):
+        has = self.has_change_permission(request, obj=obj)
+        pending = obj and obj.signoff_state == BlockSubmission.SIGNOFF_PENDING
+        return has and obj and pending and obj.can_user_signoff(request.user)
```
Comment (willdurand) [inline src/olympia/blocklist/admin.py]: The two functions above would benefit from having comments to explain the logic behind these conditions.

**Comment by eviljeff** [comment]
> The submission logs look off (it contains unrelated messages)

https://github.com/mozilla/addons-server/pull/13266/commits/146b05966c12371393c85139dbd43e3edbd83b72

File: src/olympia/blocklist/admin.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/admin.py]: https://github.com/mozilla/addons-server/issues/13278

File: src/olympia/blocklist/admin.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/blocklist/admin.py]: https://github.com/mozilla/addons-server/pull/13266/commits/61755adf93508dca052d6613741371266942c175

**Pull request merged**